http-client/src/main/java/io/airlift/http/client/jetty/BufferingResponseListener.java

```diff
@@ -15,29 +16,32 @@
 import java.nio.charset.StandardCharsets;
 import java.util.Locale;
 
+import static java.util.Objects.requireNonNull;
+
 public abstract class BufferingResponseListener
         implements Response.Listener
 {
+    private final ByteBufferAccumulator buffer;
     private final int maxLength;
-    private ByteBuffer buffer;
     private String mediaType;
     private String encoding;
 
     /**
      * Creates an instance with a default maximum length of 2 MiB.
      */
-    public BufferingResponseListener()
+    public BufferingResponseListener(ByteBufferPool byteBufferPool)
```

Comment:
Can you upstream this changes to jetty?

Reply:
https://github.com/jetty/jetty.project/issues/12687